From c4311bc8c9167a3d7011aff8728fd8067404246f Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:57 -0400
Subject: [virt] kvm/x86: expose new instruction RDSEED to guest

Message-id: <1406040016-3289-143-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86221
O-Subject: [RHEL7 PATCH v2 142/221] KVM: x86: expose new instruction RDSEED to guest
Bugzilla: 1084614 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Liu, Jinsong" <jinsong.liu@intel.com>

RDSEED instruction return a random number, which supplied by a
cryptographically secure, deterministic random bit generator(DRBG).

Signed-off-by: Xudong Hao <xudong.hao@intel.com>
Signed-off-by: Liu Jinsong <jinsong.liu@intel.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 0c79893b2bad49e0c391a9499f50fcd5b0f80874)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index 29aa323..bf8cac8 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -303,7 +303,7 @@ static inline int __do_cpuid_ent(struct kvm_cpuid_entry2 *entry, u32 function,
  /* cpuid 7.0.ebx */
  const u32 kvm_supported_word9_x86_features =
   F(FSGSBASE) | F(BMI1) | F(HLE) | F(AVX2) | F(SMEP) |
-  F(BMI2) | F(ERMS) | f_invpcid | F(RTM);
+  F(BMI2) | F(ERMS) | f_invpcid | F(RTM) | F(RDSEED);
 
  /* all calls to cpuid_count() should be made on the same cpu */
  get_cpu();
-- 
1.7.1